src/lib.rs

```diff
@@ -6,11 +6,13 @@ extern crate core as std;
 #[cfg(not(feature = "std"))]
 extern crate alloc;
 
+use core::convert::Infallible;
 use std::{
```

Comment:
Huh, why does this include `std`, even when `feature="std"` is not supplied? Looks like a bug to me...